tests/push/test_push_rule_evaluator.py

```diff
@@ -37,6 +37,30 @@
 from tests.test_utils.event_injection import create_event, inject_member_event
 
 
+class FlattenDictTestCase(unittest.TestCase):
```

Comment:
Might be worth a test case where this function is given an `EventBase`---it seems to have bespoke logic for events.

Reply:
Maybe -- I'm not changing that behavior so I'll try to do that as a follow-up so I can at least avoid the self-conflicts.